src/main/java/io/reactivex/internal/operators/mixed/ObservableConcatMapCompletable.java

```diff
@@ -0,0 +1,277 @@
+import io.reactivex.plugins.RxJavaPlugins;
+
+/**
+ * Maps the upstream intems into {@link CompletableSource}s and subscribes to them one after the
+ * other completes or terminates (in error-delaying mode).
+ * @param <T> the upstream value type
+ * @since 2.1.11 - experimental
+ */
+@Experimental
+public final class ObservableConcatMapCompletable<T> extends Completable {
+
+    final Observable<T> source;
+
+    final Function<? super T, ? extends CompletableSource> mapper;
+
+    final ErrorMode errorMode;
+
+    final int prefetch;
+
+    public ObservableConcatMapCompletable(Observable<T> source,
+            Function<? super T, ? extends CompletableSource> mapper,
+            ErrorMode errorMode,
+            int prefetch) {
+        this.source = source;
+        this.mapper = mapper;
+        this.errorMode = errorMode;
+        this.prefetch = prefetch;
+    }
+
+    @Override
+    protected void subscribeActual(CompletableObserver s) {
+        source.subscribe(new ConcatMapCompletableObserver<T>(s, mapper, errorMode, prefetch));
+    }
+
+    static final class ConcatMapCompletableObserver<T>
+    extends AtomicInteger
+    implements Observer<T>, Disposable {
+
+        private static final long serialVersionUID = 3610901111000061034L;
+
+        final CompletableObserver downstream;
+
+        final Function<? super T, ? extends CompletableSource> mapper;
+
+        final ErrorMode errorMode;
+
+        final AtomicThrowable errors;
+
+        final ConcatMapInnerObserver inner;
+
+        final int prefetch;
+
+        final SimplePlainQueue<T> queue;
+
+        Disposable upstream;
+
+        volatile boolean active;
+
+        volatile boolean done;
+
+        volatile boolean disposed;
+
+        ConcatMapCompletableObserver(CompletableObserver downstream,
+                Function<? super T, ? extends CompletableSource> mapper,
+                ErrorMode errorMode, int prefetch) {
+            this.downstream = downstream;
+            this.mapper = mapper;
+            this.errorMode = errorMode;
+            this.prefetch = prefetch;
+            this.errors = new AtomicThrowable();
+            this.inner = new ConcatMapInnerObserver(this);
+            this.queue = new SpscLinkedArrayQueue<T>(prefetch);
+        }
+
+        @Override
+        public void onSubscribe(Disposable s) {
+            if (DisposableHelper.validate(upstream, s)) {
+                this.upstream = s;
+                downstream.onSubscribe(this);
+            }
+        }
+
+        @Override
+        public void onNext(T t) {
+            queue.offer(t);
+            drain();
+        }
+
+        @Override
+        public void onError(Throwable t) {
+            if (errors.addThrowable(t)) {
+                if (errorMode == ErrorMode.IMMEDIATE) {
+                    disposed = true;
+                    inner.dispose();
+                    t = errors.terminate();
+                    if (t != ExceptionHelper.TERMINATED) {
```

Comment:
this path does not seem to be covered. same for 126. Is travis just flaky again?

Reply:
That's a probabilistic race that the CI tends to not hit, no matter the race-loop count. It gets covered 80% of the time and unfortunately, I'd have to hack the state of the operator to get it execute that inner part a second time.

Reply:
Yeah thought so. Then never mind.